ddtrace/debugging/_probe/remoteconfig.py

```diff
@@ -170,7 +129,7 @@ def update_args(cls, args, attribs):
             rate = sampling.get("snapshotsPerSecond", rate)
 
         args.update(
-            condition=_compile_expression(attribs.get("when")),
+            condition=DDExpression.compile(attribs["when"]) if "when" in attribs else None,
```

Comment:
why this one is get compiled using regular expression and not redacted expression?

Reply:
The condition should just evaluate to a boolean, no? And we wouldn't have the interim values logged anywhere, so even if we access sensitive values here we don't have a problem?

Reply:
interesting point. need to sync with other impls.